.github/workflows/node.yml

```
@@ -0,0 +1,27 @@
+name: Problem CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: Install Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '12.x'
+      
+      - name: Install Dependencies
+        run: yarn install
+      
+      - name: Run tests
+        run: yarn test
+      
+      - name: Build Problem Library
+        run: yarn build
```

We should run testing on different systems:

```suggestion
jobs:
  test-pr:
    name: Test PR - ${{ matrix.os }}
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, macos-latest, windows-latest]
    steps:
      - uses: actions/checkout@v3
      
      - name: Install Node.js
        uses: actions/setup-node@v3
        with:
          node-version: '12.x'
      
      - name: Install Dependencies
        run: yarn install
      
      - name: Run tests
        run: yarn test
      
      - name: Build Problem Library
        run: yarn build
```